game/src/pregame.rs

```diff
@@ -93,62 +93,87 @@ pub struct MainMenu {
 impl MainMenu {
     pub fn new(ctx: &mut EventCtx, app: &App) -> Box<dyn State<App>> {
         let col = vec![
-            Btn::svg_def("system/assets/pregame/quit.svg")
-                .build(ctx, "quit", Key::Escape)
-                .align_left(),
             {
                 let mut txt = Text::from(Line("A/B STREET").display_title());
                 txt.add(Line("Created by Dustin Carlino, Yuwen Li, & Michael Kirk"));
                 txt.draw(ctx).centered_horiz()
             },
-            Widget::row(vec![
-                Btn::svg(
-                    "system/assets/pregame/tutorial.svg",
-                    RewriteColor::Change(Color::WHITE, app.cs.hovering),
-                )
-                .tooltip({
-                    let mut txt = Text::tooltip(ctx, Key::T, "Tutorial");
-                    txt.add(Line("Learn how to play the game").small());
-                    txt
-                })
-                .build(ctx, "Tutorial", Key::T),
-                Btn::svg(
-                    "system/assets/pregame/sandbox.svg",
-                    RewriteColor::Change(Color::WHITE, app.cs.hovering),
-                )
-                .tooltip({
-                    let mut txt = Text::tooltip(ctx, Key::S, "Sandbox");
-                    txt.add(Line("No goals, try out any idea here").small());
-                    txt
-                })
-                .build(ctx, "Sandbox mode", Key::S),
-                Btn::svg(
-                    "system/assets/pregame/challenges.svg",
-                    RewriteColor::Change(Color::WHITE, app.cs.hovering),
-                )
-                .tooltip({
-                    let mut txt = Text::tooltip(ctx, Key::C, "Challenges");
-                    txt.add(Line("Fix specific problems").small());
-                    txt
-                })
-                .build(ctx, "Challenges", Key::C),
-            ])
+            Widget::row({
+                let btn_builder = app
+                    .cs
+                    .btn_primary_dark()
+                    .image_dims(ScreenDims::new(200.0, 100.0))
```

Comment:
I'm trying to replace all the "pictures of text" with text.

Reply:
It's satisfying and more future-proof to not bake in the text; cool! Only red flag here is `.image_dims(ScreenDims::new(200.0, 100.0))` -- I only glanced through the code, but is this specifying some resize behavior, or repeating the true dims of the SVG? If the latter, should we defer figuring out the dims till we have a `ctx` and can calculate it? If the former, what if the aspect ratio of the resize doesn't match the image?

Reply:
> but is this specifying some resize behavior, or repeating the true dims of the SVG?

Specifying the dimensions is optional. If you don't specify, it'll infer the intrinsic dimensions from the contents of the SVG. Specifying is just for resizing.

Anecdotally, I think usually the intrinsic size isn't what we want. Especially with assets copied out of figma, everything is trimmed. So one icon is 14.2x17.3 while another is 16.2x15.1, etc. We want them to be uniformly sized. 

We could, as a point of process, require that all assets are prepared for the precise resolution we want, but that seems to just be moving the effort to the design stage rather than the programming stage, and I don't think it's all that arduous for us. 

- easier if we want to tweak the design, e.g. uniformly bumping up or down all our icon sizes, without having to manually resave them
- the designs show several classes of button sizes. I haven't implemented that yet because I haven't seen it used yet, but I don't think we'd want a new svg asset for each size class (e.g. pencil20px.svg, pencil24px.svg, pencil28.svg)

So, even though it's not required if we have our asset prepared exactly how we want it, I'm pretty resigned to usually specifying the size for images. Though I hope that we can mostly centralize this specification like here: 

```
    fn btn_primary_light_icon_text(&self, image_path: &'a str, text: &'a str) -> ButtonBuilder<'a> {
        self.btn_primary_light()
            .label_text(text)
            .image_path(image_path)
            .image_dims(ScreenDims::square(18.0))
    }
```

> If the former, what if the aspect ratio of the resize doesn't match the image?

I have exposed a way to specify the content scaling rules, but the default, and the only one we're currently using anywhere is `ScaleAspectFit`. I haven't really tested the others yet, since I haven't ended up needing them, but intend to before I mark this ready for review. 

```
/// Rules for how content should stretch to fill it's bounds
#[derive(Clone, Debug)]
pub enum ContentMode {
    /// Maintaining aspect ratio, content grows until it touches its bounds in one dimension.
    /// This is the default ContentMode.
    ///
    /// If the aspect ratio of the bounds do not exactly match the aspect ratio of the content,
    /// then there will be some empty space within the bounds to center the content.
    ScaleAspectFit,

    /// Stretches content to fit its bounds exactly, breaking aspect ratio as necessary.
    ScaleToFill,

    /// Maintaining aspect ration, content grows until both bounds are met.
    ///
    /// If the aspect ratio of the bounds do not exactly match the aspect ratio of the content,
    /// the content will overflow one dimension of its bounds.
    ScaleAspectFill,
}
```

By using ScaleAspect fit, we can avoid distorting our image *and* preserve the requested amount of space so we can have a consistent layout for things like icon buttons, where the actual dimensions of the svg vary slightly depending on the icon.


What do you think?

Reply:
SGTM! By the very nature of vector graphics, the original size is pretty arbitrary, and it's indeed easier to programmatically specify this. `ScaleAspectFill` with the implicit centering makes perfect sense as a default.

Reply:
>  ScaleAspectFill with the implicit centering makes perfect sense as a default.

Great. Just to be clear, the default is ScaleAspect**Fit**, which as you describe, grows as much as it can, while staying within the bounds, centering within the unfilled bound.